src/main.rs

```diff
@@ -279,5 +295,22 @@ async fn main() -> Result<()> {
         });
     }
 
-    listen_and_serve(&args).await
+    let mut join_set = args
+        .source_addresses
+        .clone()
+        .into_iter()
+        .zip(args.destination_addresses.clone().into_iter())
+        .map(|(source_address, destination_address)| {
+            let args = args.clone();
+            tokio::spawn(async move {
+                listen_and_serve(
+                    source_address.to_string(),
+                    destination_address.to_string(),
+                    args,
+                )
+                .await
+            })
+        })
+        .collect::<JoinSet<_>>();
+    join_set.join_next().await.unwrap().unwrap().unwrap()
```

Comment:
I kinda agree with semgrep here. The current code doesn't actually exit through this path, but the unwraps are hard to follow. It also seems odd to exit after the first task without waiting for the others complete (future, hypothetical) cleanup. What about something like
```rust
// Wait until all listening tasks exit...
while let Some(task) = join_set.join_next().await {
    let _ = task.context("Listener task error")?;
    info!("Listener task completed ok");
}
Ok(())
````

Reply:
> The current code doesn't actually exit through this path, but the unwraps are hard to follow

Replaced the `unwrap`s with `expect`s to add more context on the errors that they are dealing with. They are placed there to handle conditions that wouldn't happen at all, and panics within the forwarding tasks themselves. The actual anyhow `Result` is still returned to the main function.

> What about something like

A 'successful' task will never actually complete, it will continue to forward requests indefinitely. I added a subsequent 'shutdown' call to abort all other tasks when an error is encountered.